frontend/src/components/question-creation/question-elements/SaveQuestionEditorButton.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useContext } from "react";
+import { Button } from "@chakra-ui/react";
+
+import QuestionEditorContext from "../../../contexts/QuestionEditorContext";
+import {
+  QuestionElement,
+  QuestionElementType,
+  ResponseElementType,
+} from "../../../types/QuestionTypes";
+import { updatedQuestionElement } from "../../../utils/QuestionUtils";
+
+interface SaveQuestionEditorButtonProps {
+  setShowQuestionEditor: React.Dispatch<React.SetStateAction<boolean>>;
+  setQuestions: React.Dispatch<React.SetStateAction<QuestionElement[][]>>;
+}
+
+const SaveQuestionEditorButton = ({
+  setShowQuestionEditor,
+  setQuestions,
+}: SaveQuestionEditorButtonProps): React.ReactElement => {
+  const questionError =
+    "Please create a question to be associated with this response";
+  const responseError =
+    "Please add at least one response type for this question";
+  const emptyElementError =
+    "Please ensure all fields are filled. If you do not need this component, please delete this component.";
+
+  const {
+    questionElements,
+    setQuestionElements,
+    setShowEditorError,
+  } = useContext(QuestionEditorContext);
+
+  const setElementError = (element: QuestionElement, errorText: string) => {
+    setQuestionElements((prevElements) => {
+      return updatedQuestionElement(
+        element.id,
+        element.data,
+        prevElements,
+        errorText,
+      );
+    });
+  };
+
+  const isQuestionPairError = (elementError: string) => {
+    return elementError === questionError || elementError === responseError;
+  };
+
+  const validateNoQuestionPairErrors = () => {
```

Comment:
also can you just have text / image alone without a response? bc then it creates its own question card that's blank
<img width="749" alt="image" src="https://user-images.githubusercontent.com/69697180/227792990-19938d9c-f669-4e9a-a0c5-7e6623c356e0.png">

Reply:
good catch! i'll add handling for this